pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -407,21 +407,19 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
     protected CompletableFuture<Message<T>> internalReceiveAsync() {
         CompletableFutureCancellationHandler cancellationHandler = new CompletableFutureCancellationHandler();
         CompletableFuture<Message<T>> result = cancellationHandler.createFuture();
-        Message<T> message = null;
+
         try {
-            message = incomingMessages.poll(0, TimeUnit.MILLISECONDS);
+            reentrantLock.lock();
```

Comment:
move out of the "try" block please (here and below)

Reply:
@eolivelli Thanks for your suggestion, could you explain a little bit?

Reply:
remove lock